Public/Src/Cache/ContentStore/Distributed/NuCache/DistributedCentralStorage.cs

```diff
@@ -108,13 +108,15 @@ protected override async Task<BoolResult> ShutdownCoreAsync(OperationContext con
         /// <inheritdoc />
         protected override async Task<BoolResult> TouchBlobCoreAsync(OperationContext context, AbsolutePath file, string storageId, bool isUploader)
         {
-            (var hash, var fallbackStorageId) = ParseCompositeStorageId(storageId);
+            var (hash, fallbackStorageId) = ParseCompositeStorageId(storageId);
 
             // Need to touch in fallback storage as well so it knows the content is still in use
-            await _fallbackStorage.TouchBlobAsync(context, file, fallbackStorageId, isUploader).ThrowIfFailure();
+            var touchTask = _fallbackStorage.TouchBlobAsync(context, file, fallbackStorageId, isUploader).ThrowIfFailure();
 
             // Ensure content is present in private CAS and registered
-            await PutAndRegisterFileAsync(context, file, hash);
+            var registerTask = PutAndRegisterFileAsync(context, file, hash);
+
+            await Task.WhenAll(touchTask, registerTask);

```

Comment:
Unread exceptions?

Reply:
As we discussed, WhenAll throws an error.

---
In reply to: [319297916](https://github.com/microsoft/BuildXL/pull/818#discussion_r319297916) [](ancestors = 319297916)